### PR TITLE
fix(condo): DOMA-4479 added feature flags for disabling meter notifications cron task

### PR DESCRIPTION
**Pull request by akarjakin**

### CODE REVIEW

File: apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js [hunk 1]
```diff
@@ -84,6 +87,16 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
     // * Send message with specific unique key for expired verification (key is: {meterId}_{residentId}_{now().minusMonth().format(YYYY_MM)})
     // * Print some stat
 
+    const isFeatureEnabled = await checkFeatureFlagState( SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK )
```
Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:90]:
```suggestion
    const isFeatureEnabled = await checkFeatureFlagState(SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK)
```

File: apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js [hunk 2]
```diff
@@ -84,6 +87,16 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
...
+
+    // Skip sending notifications if feature is disabled on https://growthbook.doma.ai/features
+    if (!isFeatureEnabled) {
+        logger.info({ message: `sendSubmitMeterReadingsPushNotifications was skipped due to disabled growthbook feature flag [${SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK}]` })
+
+        return
```
Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:96]:
```suggestion
        return 'disabled'
```

File: apps/condo/domains/meter/tasks/sendVerificationDateReminder.js [hunk 3]
```diff
@@ -226,6 +233,15 @@ const sendReminders = async ({ context, messages }) => {
 }
 
 const sendVerificationDateReminder = async ({ date, searchWindowDaysShift, daysCount }) => {
+    const isFeatureEnabled = await checkFeatureFlagState( SEND_METER_VERIFICATION_DATE_REMINDER_TASK )
+
+    // Skip sending notifications if feature is disabled on https://growthbook.doma.ai/features
+    if (!isFeatureEnabled) {
+        logger.info({ message: `sendVerificationDateReminder was skipped due to disabled growthbook feature flag [${SEND_METER_VERIFICATION_DATE_REMINDER_TASK}]` })
```
Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendVerificationDateReminder.js:240]:
```suggestion
        logger.info(`sendVerificationDateReminder was skipped due to disabled growthbook feature flag [${SEND_METER_VERIFICATION_DATE_REMINDER_TASK}]`)
```

File: apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js [hunk 4]
```diff
@@ -84,6 +87,16 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
     // * Send message with specific unique key for expired verification (key is: {meterId}_{residentId}_{now().minusMonth().format(YYYY_MM)})
     // * Print some stat
 
+    const isFeatureEnabled = await checkFeatureFlagState( SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK )
+
+    // Skip sending notifications if feature is disabled on https://growthbook.doma.ai/features
+    if (!isFeatureEnabled) {
+        logger.info({ message: `sendSubmitMeterReadingsPushNotifications was skipped due to disabled growthbook feature flag [${SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK}]` })
```
Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:94]:
```suggestion
        logger.info(`sendSubmitMeterReadingsPushNotifications was skipped due to disabled growthbook feature flag [${SEND_SUBMIT_METER_READINGS_PUSH_NOTIFICATIONS_TASK}]`)
```

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.js [hunk 5]
```diff
@@ -10,13 +10,12 @@ const { sendResidentsNoAccountNotifications } = require('./sendResidentsNoAccoun
...
 const sendBillingReceiptNotifications = async () => {
-    const data = { req: { features: await featureToggleManager.fetchFeatures() } }
-    const isFeatureEnabled = featureToggleManager.isFeatureEnabled( data, SEND_BILLING_RECEIPTS_NOTIFICATIONS_TASK )
+    const isFeatureEnabled = await checkFeatureFlagState( SEND_BILLING_RECEIPTS_NOTIFICATIONS_TASK )
 
     // Skip sending notifications if feature is disabled on https://growthbook.doma.ai/features
     // This affects only cron task, notifications still could be sent using scripts in condo/
     if (!isFeatureEnabled) {
-        logger.info({ message: `sendBillingReceiptNotifications cron task skipped due to disabled growthbook feature flag [${SEND_BILLING_RECEIPTS_NOTIFICATIONS_TASK}]` })
+        logger.info({ message: `sendBillingReceiptNotifications was skipped due to disabled growthbook feature flag [${SEND_BILLING_RECEIPTS_NOTIFICATIONS_TASK}]` })
```
Comment (pahaz) [inline apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.js:18]:
```suggestion
        logger.info(`sendBillingReceiptNotifications was skipped due to disabled growthbook feature flag [${SEND_BILLING_RECEIPTS_NOTIFICATIONS_TASK}]`)
```

File: packages/featureflags/featureToggleManager.js [hunk 6]
```diff
@@ -73,6 +73,13 @@ class FeatureToggleManager {
 
 const featureToggleManager = new FeatureToggleManager()
 
+const checkFeatureFlagState = async (flagKey) => {
```
Comment (pahaz) [inline packages/featureflags/featureToggleManager.js:76]:
```suggestion
const checkFeatureFlagState = async (featureName, featuresContext) => {
```

File: apps/condo/domains/meter/tasks/sendVerificationDateReminder.spec.js [hunk 7]
```diff
@@ -5,104 +5,122 @@
...
 describe('Meter verification notification', () => {
     setFakeClientMode(index)
 
-    it('should not send messages on null nextVerificationDate', async () => {
-        const messages = await getNotificationsFromMeter({
-            verificationDate: null,
-            nextVerificationDate: null,
+    describe('Direct function tests', () => {
+        it('should not send messages on null nextVerificationDate', async () => {
+            const messages = await getNotificationsFromMeter({
```
Comment (dkoviazin) [inline apps/condo/domains/meter/tasks/sendVerificationDateReminder.spec.js:33]: You need a test if verificationDate is set and nextVerificationDate is null

File: apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js [hunk 8]
```diff
@@ -255,6 +257,20 @@ const sendMessagesForExpiredMeterVerificationDate = async ({ context, metersWith
     }))
 }
 
+const sendSubmitMeterReadingsPushNotificationsTaskFn = async (context = null) => {
```
Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:260]: move this logic inside isFeatureEnabled

Comment (pahaz) [inline apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:260]: it's too many wrappers. Could you please move all this logic inside the apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotificationsTask.js

**Review by pahaz — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js (hunk 8 above)
Comment (akarjakin) [reply apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js:260]: fixed

File: apps/condo/domains/meter/tasks/sendVerificationDateReminder.spec.js (hunk 7 above)
Comment (akarjakin) [reply apps/condo/domains/meter/tasks/sendVerificationDateReminder.spec.js:33]: done

**Review by dkoviazin — APPROVED** [review]
